### ISSUE TITLE
Error: Cannot find module 'fs'

### ISSUE THREAD

**Issue by dimakaplin** [Keyang/node-csvtojson#285]
Hello! I use npm only without webpack and other. I need to use csvtojson to convert csv in my app, but in dont works. I become this error: 

Error: Cannot find module 'fs'
▶ 2 stack frames were collapsed.
App.componentDidMount
src/App.js:43
  40 | componentDidMount() {
  41 | 
  42 |  // this.twoPies();
> 43 |  csv()
     | ^  44 |  .fromFile(csvFilePath)
  45 |  .then((jsonObj)=>{
  46 |      console.log(jsonObj);

its part of my code:

const csv = require('csvtojson');
const csvFilePath = '/data.csv';
/............/
   csv()
   .fromFile(csvFilePath)
   .then((jsonObj)=>{
       console.log(jsonObj);
   })

**Comment by Keyang** [comment]
do you use the lib in node.js or in browser?

**Comment by dimakaplin** [comment]
yes, of course i use node js v 8.10.0, app was created with create-react-app

**Comment by Keyang** [comment]
Ok you are building a react app so you cannot use `.fromFile` as you wont be able to access filesystem from a browser..

**Comment by dldugan14** [comment]
I'm getting that same error in a react app in electron where i do have access to fs any ideas?

**Comment by Graceliying82** [comment]
Same issue here.

**Comment by Unionindesign** [comment]
the npm package Papaparse works well! I was getting the same error with csvtojson about .fromFile - you can combine FileReader with csv to JSON, or papaparse already has filereader built in. Still trying to format my data correctly but happy to post some code if anyone wants to see it!

Well what the heck, here's what I have so far, the csv data is logging correctly in the console as an object so off to a good start!

FYI don't necessarily need to run this in componentDidMount is using react...I intend to fetch the data in componentDidMount once we have it encrypted correctly, you might also use componentWillMount etc.
class DataHandler extends Component {
	constructor(props) {
		super(props);
		this.state = {
			data: null
		};
//bind your callback method.....
		this.updateData = this.updateData.bind(this);
	}

`	componentDidMount() {           
		const Papa = require("papaparse");
		const dataFilePath = require("../../data/myDataFile.csv");
		Papa.parse(dataFilePath, {
			download: true,
			header: true,
			//transformHeader: undefined,
			delimiter: ",",
			newline: ",",
			//escapeChar: '"',
			complete: this.updateData  <--   (Papa.parse() does not return anything! you need a callback like below)
		});
	}
	updateData(result) {
		const data = result.data;
		console.log(data);`

**Comment by RushhourPark** [comment]
I have a similar problem with the "Cannot find module 'fs'" error. I used csvtojson in browser by embedding the 'csvtojson.min.js' directly into script tag, then the colde:
`csv({
	output: 'json'
})
.fromFile(csvFilePath)
.then(function(result){
	console.log(result);
})`

gave the error of cannot find module 'fs'.

**Comment by geomC** [comment]
Guys, [`fs`](https://nodejs.org/api/fs.html) is only available in a node environment, if you run javascript on a server, e.g. an express app.

If you use this library in a browser, `fs` is not available.

The are other libraries for parsing a csv to json in javascript that runs a browser, e.g. [papaparse](https://www.papaparse.com/).

**Comment by dimslaev** [comment]
Then why it says in the README it has support for browser?

**Comment by kriscarle** [comment]
For anyone else that ends up here looking for an in-browser example, this is what worked for me:

```js
import {csv} from "csvtojson";
// ...
const res = await fetch(url);
const text = await res.text();
const jsonArray = await csv().fromString(text);
```
couple of notes:

* You may need to polyfill `fetch()` to support older browsers, or using something else to actually download the CSV and turn it into a string, or use something else, there are lots of options: 'superagent', 'axios', 'jquery', etc.
* Note the import is different from the current example in the README, if you `import * as csv` then you'll need to use `csv.csv()` because it imports an object, not the csv function

**Comment by 777akot** [comment]
in browser you can read csv with filereader and use fromString with result

**Comment by mebibou** [comment]
For those that use Angular 19 for example, you can do:
```typescript
import csvtojson from 'csvtojson';

await csvtojson().fromString(data);
```
